reverted:
--- b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -5204,11 +5204,11 @@
 	caps = &dm->backlight_caps[aconnector->bl_idx];
 	if (get_brightness_range(caps, &min, &max)) {
 		if (power_supply_is_system_supplied() > 0)
+			props.brightness = DIV_ROUND_CLOSEST((max - min) * caps->ac_level, 100);
-			props.brightness = DIV_ROUND_CLOSEST(max * caps->ac_level, 100);
 		else
+			props.brightness = DIV_ROUND_CLOSEST((max - min) * caps->dc_level, 100);
-			props.brightness = DIV_ROUND_CLOSEST(max * caps->dc_level, 100);
 		/* min is zero, so max needs to be adjusted */
+		props.max_brightness = max - min;
-		props.max_brightness = max;
 		drm_dbg(drm, "Backlight caps: min: %d, max: %d, ac %d, dc %d\n", min, max,
 			caps->ac_level, caps->dc_level);
 	} else
